### PR TITLE
[NFC] SILGlobalVariable utilities.

### PR DESCRIPTION
**Pull request by atrick**
Ignore the first two commits, which are from https://github.com/apple/swift/pull/16834.

The third commit just moves some code to a place it belongs: Move the utilities that are required for recognizing SILGlobalVariable access into SILGlobalVariable.[h|cpp].

Should be a trivial review. I'm not making a serious attempt to rework the interfaces:
@eeckstein, @gottesmm, @shajrawi ?

### CODE REVIEW

File: lib/SIL/SILGlobalVariable.cpp [hunk 1]
```diff
@@ -125,3 +127,132 @@ ClangNode SILGlobalVariable::getClangNode() const {
...
+
+  // If the globalinit_func is trivial, continue; otherwise bail.
+  SingleValueInstruction *dummyInitVal;
+  auto *SILG = getVariableOfStaticInitializer(InitF, dummyInitVal);
+
+  return SILG;
+}
+
+SILFunction *swift::getCalleeOfOnceCall(BuiltinInst *BI) {
+  assert(BI->getNumOperands() == 2 && "once call should have 2 operands.");
```
Comment (shajrawi) [inline lib/SIL/SILGlobalVariable.cpp]: Can you please assert based on `getBuiltinInfo().ID` of the `BuiltinInst` ? cleaner and safer assertion.

File: lib/SIL/SILGlobalVariable.cpp [hunk 2]
```diff
@@ -125,3 +127,132 @@ ClangNode SILGlobalVariable::getClangNode() const {
...
+    return nullptr;
+  return getCalleeOfOnceCall(CallToOnce);
+}
+
+SILGlobalVariable *
+swift::getVariableOfStaticInitializer(SILFunction *InitFunc,
+                                      SingleValueInstruction *&InitVal) {
+  InitVal = nullptr;
+  SILGlobalVariable *GVar = nullptr;
+  // We only handle a single SILBasicBlock for now.
```
Comment (shajrawi) [inline lib/SIL/SILGlobalVariable.cpp]: I see this comment and this check repeated here and in `findInitializer`, maybe we can outline it to a single `static bool` function?

**Review by shajrawi — COMMENTED** [review]
Overall LGTM! but see the couple of suggestions I made in comments :)

**Comment by atrick** [comment]
@swift-ci smoke test.

**Pull request merged**